### DISCUSSION TITLE
Pip psutil wont install

### DISCUSSION THREAD

**Discussion by ItzDaweCZE** [termux/termux-packages#17929] [Q&A] [answered]
I use termux on my phone and when I do "pip install psutil" it says this please help:

 $ pip install psutil
Collecting psutil
  Using cached psutil-5.9.5.tar.gz (493 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: psutil
  Building wheel for psutil (pyproject.toml) ...error
  error: subprocess-exited-with-error

  × Building wheel for psutil (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [45 lines of output]
      CANNOT LINK EXECUTABLE "/data/data/com.termux/files/usr/bin/cc": cannot locate symbol "__emutls_get_address" referenced by "/data/data/com.termux/files/usr/lib/libclang-cpp.so"...
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-aarch64-cpython-311
      creating build/lib.linux-aarch64-cpython-311/psutil
      copying psutil/__init__.py -> build/lib.linux-aarch64-cpython-311/psutil
      copying psutil/_common.py -> build/lib.linux-aarch64-cpython-311/psutil
      copying psutil/_compat.py -> build/lib.linux-aarch64-cpython-311/psutil
      copying psutil/_psaix.py -> build/lib.linux-aarch64-cpython-311/psutil
      copying psutil/_psbsd.py -> build/lib.linux-aarch64-cpython-311/psutil
      copying psutil/_pslinux.py -> build/lib.linux-aarch64-cpython-311/psutil
      copying psutil/_psosx.py -> build/lib.linux-aarch64-cpython-311/psutil
      copying psutil/_psposix.py -> build/lib.linux-aarch64-cpython-311/psutil
      copying psutil/_pssunos.py -> build/lib.linux-aarch64-cpython-311/psutil
      copying psutil/_pswindows.py -> build/lib.linux-aarch64-cpython-311/psutil
      creating build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/__init__.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/__main__.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/runner.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_aix.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_bsd.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_connections.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_contracts.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_linux.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_memleaks.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_misc.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_osx.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_posix.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_process.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_sunos.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_system.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_testutils.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_unicode.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      copying psutil/tests/test_windows.py -> build/lib.linux-aarch64-cpython-311/psutil/tests
      running build_ext
      building 'psutil._psutil_linux' extension
      creating build/temp.linux-aarch64-cpython-311
      creating build/temp.linux-aarch64-cpython-311/psutil
      aarch64-linux-android-clang -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-strong -O3 -fstack-protector-strong -O3 -fPIC -DPSUTIL_POSIX=1 -DPSUTIL_SIZEOF_PID_T=4 -DPSUTIL_VERSION=595 -DPy_LIMITED_API=0x03060000 -DPSUTIL_ETHTOOL_MISSING_TYPES=1 -DPSUTIL_LINUX=1 -I/data/data/com.termux/files/usr/include/python3.11 -c psutil/_psutil_common.c -o build/temp.linux-aarch64-cpython-311/psutil/_psutil_common.o
      CANNOT LINK EXECUTABLE "/data/data/com.termux/files/usr/bin/aarch64-linux-android-clang": cannot locate symbol "__emutls_get_address" referenced by "/data/data/com.termux/files/usr/lib/libclang-cpp.so"...
      psutil could not be installed from sources. Perhaps Python header files are not installed. Try running:
        sudo apt-get install gcc python3-dev
      error: command '/data/data/com.termux/files/usr/bin/aarch64-linux-android-clang' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for psutil
Failed to build psutil
ERROR: Could not build wheels for psutil, which is required to install pyproject.toml-based projects
$

**Comment by licy183** [comment]
Please update all your packages to latest. `pkg up`

**Comment by sylirre** [comment, marked as answer]
`pip3 install psutil` works, no errors reproducible.

Upgrade your packages.

**Comment by 0000p0** [comment]
# Install clang-16 from tur-repo
pkg install clang-16

# Set environment variable before pip install
export CC=clang-16
pip install aiohttp

**Reply by sleepysoong** [reply]
You saved me. Thank You!

**Comment by mdm33333333-byte** [comment]
Not working

**Reply by TomJo2000** [reply]
Please try installing the `python-psutil` package.
```bash
pkg up
pkg i python-psutil
```

**Reply by ChikoRenaldo** [reply]
Thx it's work
